### PR TITLE
Widget Improvements

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- New centralised `make_any_selector` for creating dropdown for various types.
- Updated `Filter_Condition.default_widget` to be auto-scoped.
image
- Nested structure for Simple_Expressions.
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso [hunk 1]
```diff
@@ -94,45 +94,41 @@ make_column_name_multi_selector table display:Display=Display.Always add_regex:B
...
-make_fill_default_value_selector : Table | Nothing -> Display -> Boolean -> Boolean -> Boolean -> Widget
-make_fill_default_value_selector column_source=Nothing display=Display.Always add_text:Boolean=False add_number:Boolean=False add_boolean:Boolean=False =
-    column_ref = make_column_ref_by_name_selector column_source display add_text add_regex=False add_number=add_number add_boolean=add_boolean
+make_fill_default_value_selector column_source=Nothing display:Display=..Always value_type:Value_Type=Value_Type.Mixed add_nothing:Boolean=False =
+    add_number = value.is_numeric || value_type==Value_Type.Mixed
+    add_boolean = value_type.is_boolean || value_type==Value_Type.Mixed
+    add_text = value_type.is_text || value_type==Value_Type.Mixed
+    add_date = value_type.is_date || self.value_type==Value_Type.Mixed
+    add_time = value_type.is_time || self.value_type==Value_Type.Mixed
+    add_date_time = value_type.is_date_time || self.value_type==Value_Type.Mixed
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso]: Finally! So cool

File: distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso [hunk 2]
```diff
@@ -2897,7 +2897,7 @@ type Table
 
              fill_nothing = table.fill_nothing ["col0", "col1"] 20.5
     @columns (Widget_Helpers.make_column_name_multi_selector add_regex=True add_by_type=True)
-    @default (self -> Widget_Helpers.make_fill_default_value_selector column_source=self add_text=True add_number=True add_boolean=True)
+    @default (self-> Widget_Helpers.make_fill_default_value_selector value_type=Value_Type.Mixed add_nothing=True)
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso]: shouldn't this be `self.value_type` then?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Simple_Expression.enso [hunk 3]
```diff
@@ -91,30 +188,138 @@ type Simple_Expression
...
+    ## Perform a Text Operation
+    Text (operation : Text_Operation = ..Left 1)
+
+    ## Perform a Math Operation
+    Math (operation : Math_Operation = ..Subtract 0)
+
+    ## Perform a Date Operation
+    Date (operation : Date_Operation = ..Date_Add 1)
+
+    ## Perform a Date Operation
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Simple_Expression.enso]: typo

```suggestion
    ## Perform a Logical Operation
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso]: Can't wait to also see columns filtered by type in `aggregate` :)

**Review by radeusgd — APPROVED** [review]
Looks good. One typo to fix.

Shouldn't we now use `self.value_type` in `fill_nothing` instead of `Mixed`?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso]: Not on a table - am trying to think how best to handle it.

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso]: Oh, my oversight...

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**